tools/docker/run.sh

```diff
@@ -42,7 +42,7 @@ generate_container_random_ip() {
 }
 
 gateway_netid_length() {
-    docker network inspect prplMesh-net --format "{{(index .IPAM.Config 0).Subnet}}" | sed -rn 's/^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}(\/[0-9]{2})$/\1/p'
+    docker network inspect prplMesh-net-${SUDO_USER:-${USER}} --format "{{(index .IPAM.Config 0).Subnet}}" | sed -rn 's/^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}(\/[0-9]{2})$/\1/p'
```

Comment:
instead of the username, it makes more sense to pass in a unique ID as an argument (which could default to `${SUDO_USER:-${USER}}`. With that, we can get rid of the docker-in-docker approach and instead run it directly in a shell runner. That would speed up our run-tests CI job significantly because the cache can be used (now it re-downloads the images every time).

If adding the argument is too much work, please at least make it a variable e.g. CONTAINER_NAME_SUFFIX.

Reply:
I should probably already know this - but what do you mean by the docker-in-docker approach?

Reply:
Because the tests cannot be run in parallel on the same host at the moment, the CI runs the prplmesh-runner containers in a docker container (a new docker daemon is run inside a docker container). It works, but it's a bit overkill and has a few downsides like not keeping the cache and images so we would like to use something else (either a shell runner, or keeping the docker runner but with a single daemon).

Reply:
Understood.

Reply:
Done